### DISCUSSION TITLE
Custom variable

### DISCUSSION THREAD

**Discussion by hidech** [webstudio-is/webstudio-community#55] [Ideas]
It's wonderful if I can define custom variable and use it. How do you think of it?
It would be nice if it is not secure. Only opened value will be stored.

Case 1 : Use as defined const value, such as using many times in my JavaScript.

Case 2 : Use as passing data among the pages, and use in JavaScript.

Case 3 : Use as store user's selection of default value such as language, preferred settings.

**Comment by kof** [comment]
We have been discussing a UI for variables where you can choose a type and use it in style panel or with logical expressions. It will be also interesting to give access to them via custom code for sure!

**Comment by TrySound** [comment]
Data bindings are live
https://www.youtube.com/watch?v=2MrkfnjYuCo

**Reply by hidech** [reply]
Data bindings is great.
Is there a way to modify a variable according to user's interaction? It looks constants for now.

**Reply by kof** [reply]
that would be actions, a bit later ....

**Reply by hidech** [reply]
Thanks for your information!
